src/connectors/range/__tests__/connectRange-test.js

```diff
@@ -1283,3 +1036,287 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/range-input
     });
   });
 });
+
+describe('getWidgetState', () => {
+  test('returns the `uiState` empty', () => {
+    const render = jest.fn();
+    const makeWidget = connectRange(render);
+    const helper = jsHelper({}, 'indexName');
+    const widget = makeWidget({
+      attribute: 'price',
+    });
+
+    const actual = widget.getWidgetState(
+      {},
+      {
+        searchParameters: helper.state,
+      }
+    );
+
+    expect(actual).toEqual({});
+  });
+
+  test('returns the `uiState` with a refinement', () => {
+    const render = jest.fn();
+    const makeWidget = connectRange(render);
+    const helper = jsHelper({}, 'indexName', {
+      disjunctiveFacets: ['price'],
+      numericRefinements: {
+        price: {
+          '<=': [1000],
+          '>=': [100],
+        },
+      },
+    });
+    const widget = makeWidget({
+      attribute: 'price',
+    });
+
+    const actual = widget.getWidgetState(
+      {},
+      {
+        searchParameters: helper.state,
+      }
+    );
+
+    expect(actual).toEqual({
+      range: {
+        price: '100:1000',
```

Comment:
We could add more tests to cover all the possible usage: `x:`, `:x`, `x:x`.

Reply:
0e606b0ac0fb28e8b280981786e89326d4ba2953